File: plot_multiple_lines.m
% Generate a plot with multiple lines and save with a random filename

% Create x values
x = 0:0.1:10;

% Create multiple y values
y1 = sin(x);
y2 = cos(x);
y3 = sin(x).*cos(x);
y4 = sin(x).^2;

% Plot all lines
figure;
plot(x, y1, 'r-', 'LineWidth', 2); hold on;
plot(x, y2, 'b--', 'LineWidth', 2);
plot(x, y3, 'g:', 'LineWidth', 2);
plot(x, y4, 'k-.', 'LineWidth', 2);

% Add labels and title
xlabel('X Values');
ylabel('Y Values');
title('Multiple Lines Plot');
legend('sin(x)', 'cos(x)', 'sin(x)cos(x)', 'sin^2(x)', 'Location', 'best');
grid on;

% Generate random filename
random_name = ['plot_' num2str(randi(9999)) '.png'];

% Save the figure
saveas(gcf, random_name);

fprintf('Plot saved as %s\n', random_name);

File: plot_multiple_lines.py
import numpy as np
import matplotlib.pyplot as plt
import random
import string

# Generate x values
x = np.linspace(0, 10, 100)

# Create multiple y values
y1 = np.sin(x)
y2 = np.cos(x)
y3 = np.sin(x) * np.cos(x)
y4 = np.sin(x)**2

# Plot all lines
plt.figure(figsize=(10, 6))
plt.plot(x, y1, 'r-', linewidth=2, label='sin(x)')
plt.plot(x, y2, 'b--', linewidth=2, label='cos(x)')
plt.plot(x, y3, 'g:', linewidth=2, label='sin(x)cos(x)')
plt.plot(x, y4, 'k-.', linewidth=2, label='sin²(x)')

# Add labels and title
plt.xlabel('X Values')
plt.ylabel('Y Values')
plt.title('Multiple Lines Plot')
plt.legend(loc='best')
plt.grid(True)

# Generate random filename with letters and numbers
random_name = 'plot_' + ''.join(random.choices(string.ascii_lowercase + string.digits, k=8)) + '.png'

# Save the figure
plt.savefig(random_name)
plt.show()

print(f'Plot saved as {random_name}')